Core/Newsletter2Go/Components/Api/Resource/NewsletterCustomer.php

```diff
@@ -160,73 +123,66 @@ public function getList(
      * @return mixed
      *
      * @throws \Shopware\Components\Api\Exception\PrivilegeException
+     * @throws \Shopware\Components\Api\Exception\OrmException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     * @throws \Doctrine\ORM\ORMInvalidArgumentException
+     * @throws \InvalidArgumentException
      */
     public function update($email, array $params)
     {
         $this->checkPrivilege('update');
 
         if (empty($email)) {
-            return Nl2go_ResponseHelper::generateErrorResponse(
-                'email-param is missing',
-                Nl2go_ResponseHelper::ERRNO_PLUGIN_OTHER
-            );
+            throw new \InvalidArgumentException('email-param is missing');
         }
-        try {
-            if ($params['Unsubscribe']) {
-                /** @var $article Address */
-                $subscription = $this->getRepositoryAddress()->findOneBy(array('email' => $email));
-                $this->getManager()->remove($subscription);
-                $this->flush();
 
-                return true;
-            }
+        if ($params['Unsubscribe']) {
+            /** @var $article Address */
+            $subscription = $this->getRepositoryAddress()->findOneBy(['email' => $email]);
+            $this->getManager()->remove($subscription);
+            $this->flush();
 
-            if ($params['Subscribe']) {
-                $groups = Shopware()->Models()->getRepository(Group::class)->findAll();
-                if (empty($groups) === false && is_array($groups)) {
-                    $group = reset($groups);
-                    $groupId = $group->getId();
+            return true;
+        }
 
-                    $customer = $this->getRepositoryCustomer()->findOneBy(array('email' => $email));
-                    $subscription = new Address();
+        if ($params['Subscribe']) {
+            $groups = Shopware()->Models()->getRepository(Group::class)->findAll();
+            if (empty($groups) === false && is_array($groups)) {
+                $group = reset($groups);
```

Comment:
Just for clearification: It is okay to always use the first group?

Reply:
IDK, the only change I did there is that I removed the try-catch block - the code inside stayed the same.